FlickrApp/app/src/main/java/com/example/flickrapp/FlickrDataRepository.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.flickrapp
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import retrofit2.Call
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class FlickrDataRepository {
+
+    val retrofit: Retrofit =
+        Retrofit.Builder().baseUrl("https://www.flickr.com/services/feeds/")
+            .addConverterFactory(GsonConverterFactory.create()).build()
```

Comment:
all of this building is ok for a single api app, but if you have multiple endpoints like most apps, you will have a lot of duplicated code. Most people use dagger to create this, and then an Interface with the different methods and endpoints. It saves a lot of boilerplate and is a lot cleaner.